cg/cli/workflow/balsamic/store.py

```diff
@@ -24,123 +29,83 @@ def store(context):
-
-    return new_analysis
-
-
-def _include_files_in_housekeeper(bundle_obj, context, hk_api, version_obj):
-    """Function to include files in housekeeper"""
-    try:
-        hk_api.include(version_obj)
-    except hk.VersionIncludedError as error:
+    except AnalysisDuplicationError:
+        click.echo(click.style("analysis version already added", fg="yellow"))
+        context.abort()
+    except VersionIncludedError as error:
         click.echo(click.style(error.message, fg="red"))
         context.abort()
-    hk_api.add_commit(bundle_obj, version_obj)
+
+    status.add_commit(new_analysis)
+    click.echo(click.style("included files in Housekeeper", fg="green"))
 
 
-def _add_new_complete_analysis_record(bundle_data, family_obj, status, version_obj):
-    """Function to create and return a new analysis database record"""
-    pipeline = family_obj.links[0].sample.data_analysis
-    pipeline = pipeline if pipeline else "balsamic"
+@store.command("generate-deliverables-file")
+@click.option("-d", "--dry-run", "dry", is_flag=True, help="print command to console")
+@click.option("--config", "config_path", required=False, help="Optional")
+@click.argument("case_id")
+@click.pass_context
+def generate_deliverables_file(context, dry, config_path, case_id):
+    """Generate a deliverables file for the case_id."""
 
-    if status.analysis(family=family_obj, started_at=version_obj.created_at):
-        raise AnalysisDuplicationError(
-            f"Analysis object already exists for {family_obj.internal_id}{version_obj.created_at}"
-        )
+    conda_env = context.obj["balsamic"]["conda_env"]
+    root_dir = Path(context.obj["balsamic"]["root"])
 
-    new_analysis = status.add_analysis(
-        pipeline=pipeline,
-        version=bundle_data["pipeline_version"],
-        started_at=version_obj.created_at,
-        completed_at=dt.datetime.now(),
-        primary=(len(family_obj.analyses) == 0),
-    )
-    new_analysis.family = family_obj
-    return new_analysis
+    case_obj = context.obj["db"].family(case_id)
 
+    if not case_obj:
+        click.echo(click.style(f"Case {case_id} not found", fg="yellow"))
 
-def _reset_action_from_running_on_family(family_obj):
-    family_obj.action = None
+    if not config_path:
+        config_path = Path.joinpath(root_dir, case_id, case_id + ".json")
 
+    # Call Balsamic
+    command_str = f" plugins deliver" f" --sample-config {config_path}"
 
-def _add_new_analysis_to_the_status_api(bundle_obj, status):
-    family_obj = status.family(bundle_obj.name)
-    return family_obj
+    command = [f"bash -c 'source activate {conda_env}; balsamic"]
+    command_str += "'"
+    command.extend(command_str.split(" "))
 
+    if dry:
+        click.echo(" ".join(command))
+        return SUCCESS
 
-@store.command()
-@click.pass_context
-def completed(context):
-    """Store all completed analyses."""
-    hk_api = context.obj["hk_api"]
+    process = subprocess.run(" ".join(command), shell=True)
```

Comment:
Is `shell=True` required? 

"Executing shell commands that incorporate unsanitized input from an untrusted source makes a program vulnerable to shell injection, a serious security flaw which can result in arbitrary command execution. For this reason, the use of shell=True is strongly discouraged in cases where the command string is constructed from external input:" from the [docs](https://docs.python.org/2/library/subprocess.html#frequently-used-arguments)

Reply:
I'll test and see

Reply:
If there is a 'balsamic' alias that we want to run, then I guess this will not work without `shell=True`.

Reply:
tried and didn't get it to work as I wanted :( Can we keep it as is for now?

Reply:
Yes, I think you can keep it. But might be worth thinking about this in the future.